public/app/features/explore/ExplorePaneContainer.tsx

```diff
@@ -64,16 +69,49 @@ class ExplorePaneContainerUnconnected extends React.PureComponent<Props> {
     };
   }
 
-  componentDidMount() {
+  async componentDidMount() {
     const { initialized, exploreId, initialDatasource, initialQueries, initialRange, panelsState } = this.props;
     const width = this.el?.offsetWidth ?? 0;
-
     // initialize the whole explore first time we mount and if browser history contains a change in datasource
     if (!initialized) {
+      let queriesDatasourceOverride = undefined;
```

Comment:
Is it possible to move this initialization logic to `initializeExplore()`? It'd be easier to test and would keep the component layer a bit simpler.

Reply:
I'm going to put this in a followup ticket, https://github.com/grafana/grafana/issues/54533 

This is possible to do now but I'd rather tackle it with any other issues people find.